test/es6/function.name.js

```diff
@@ -750,6 +750,17 @@ var tests = [
             class C { foo(){} };
             assert.areEqual("foo",(new C).foo.name);
         }
+    },
+	{
+        name: "Getter and setter have correct name in defineProperty",
+        body: function()
+        {
+            var obj = {};
+            Object.defineProperty(obj, 'test', {get : function () {}, set : function () {} });

```

Comment:
should your test functions have names themselves, so the logic is explicit about which name gets priority?